nrf52_layout.ld

```diff
@@ -0,0 +1,11 @@
+/* Layout for the nRF52-DK, used by the examples in this repository. */
+
+MEMORY {
+  /* Guess: the Tock headers take 30 bytes. */
```

Comment:
Oh wait that wouldn't work, as the kernel would pass a text_start that points at the start of the header's padding rather than the start of the application's runtime header.

Perhaps we can create a new "padding" TLV type in the binary format and use that to pad the header as a whole?

Reply:
Oh I didn't catch that you wrote "given program name", I thought you just said "given program". If all you're asking for is for the linker script to compensate for a varying program name, then that seems reasonable to me.

Reply:
I think, it's the program name only. But I am not entirely sure. Maybe you could check how `elf2tbf` works.

Reply:
In general, the TBF header size is allowed to vary. However, most applications will only have the TBF header, main header, and application name. The application name is the only variable-length field that is common.

We could also pad the size of the application name field. It's already length-delimited and padded to a multiple of 4 bytes; we could instead make that user-defined.

Reply:
I discussed options with Brad Campbell and went with the "modify elf2tab so that it pads the header out to a specified size" idea. That is implemented in https://github.com/tock/elf2tab/pull/6; this PR will need to wait until that PR is merged.

Reply:
tock/elf2tab#6 has been merged.